### DISCUSSION TITLE
Feature Request: Hoard Button like the Pinterest Button

### DISCUSSION THREAD

**Discussion by w4t3bzrv** [karakeep-app/karakeep#338] [Ideas]
With Pinterest, you can install the browser extension to save an image in your board. It appears on the top left corner and looks like this. 
![image](https://github.com/user-attachments/assets/bc100cb0-4085-47c4-a58b-b4af1a625c8a)
It would be interesting to have the same feature, a black Hoard button overlayed on the top right corner to allow us to save an image into a hoard list. Once the hoard list is selected, you get to add tags, notes and keep the hotlink. It would save the high resolution source of the image locally. This feature would allow user to create a self-hosted Pinterest-alternative.

**Comment by MohamedBassem** [comment]
I've just released a new extension version that includes an "Add to Hoarder" button when you right click on images (@kamtschatka implemented it). I think this is probably good enough?
<img width="302" alt="Screenshot 2024-08-18 at 11 21 49 PM" src="https://github.com/user-attachments/assets/15971140-72ac-4620-a9ac-cff52281d0fb">

**Comment by w4t3bzrv** [comment]
I see the new 'Add to Hoarder' option in the right-click menu.
![image](https://github.com/user-attachments/assets/d0c71344-8ce3-4f97-b700-50406b31ae7e)

However, the browser extension is unable to connect to my Hoarder server, even if I have the webpage open and can manually add things to the boards. 

Is there an issue with the extension?
Also, why does the extension need access to my browser history?

**Comment by MohamedBassem** [comment]
Sorry I missed your comment. Are you saying that the extension doesn't work at all for you, or is it just the `Add to Hoarder` button? If the extension doesn't work at all, feel free to open an issue and we can see what's wrong there.

As for the extension requesting permission to access browser history, we don't. This is the the permissions that we request (https://github.com/hoarder-app/hoarder/blob/main/apps/browser-extension/manifest.json#L30):
1. Storage: To store the API keys.
2. Tabs: To access the currently active tab and get its URL to save it.
3. Context menu to add the `Add to Hoarder` button.

Chrome showing that it needs access to browser history is kinda weird.

**Comment by w4t3bzrv** [comment]
The extension seems to work but I can't make Hoarder work through Docker Desktop for Windows. I am new to this and I don't understand much of it. I already have something running on port 3000 and can't find how to change the port easily for Hoarder.
